storage/pruning/pruningStorer.go

```diff
@@ -0,0 +1,484 @@
+package pruning
+
+import (
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"regexp"
+	"sync"
+
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/data"
+	"github.com/ElrondNetwork/elrond-go/epochStart/notifier"
+	"github.com/ElrondNetwork/elrond-go/logger"
+	"github.com/ElrondNetwork/elrond-go/storage"
+	"github.com/ElrondNetwork/elrond-go/storage/storageUnit"
+)
+
+var log = logger.GetOrCreate("storage/pruning")
+
+// DefaultEpochDirectoryName represents the naming pattern for epoch directories
+const DefaultEpochDirectoryName = "Epoch"
+
+// persisterData structure is used so the persister and its path can be kept in the same place
+type persisterData struct {
+	persister storage.Persister
+	path      string
+	isClosed  bool
+}
+
+// PruningStorer represents a storer which creates a new persister for each epoch and removes older activePersisters
+type PruningStorer struct {
+	lock                  sync.RWMutex
+	fullArchive           bool
+	activePersisters      []*persisterData
+	persistersMapByEpoch  map[uint32]*persisterData
+	cacher                storage.Cacher
+	bloomFilter           storage.BloomFilter
+	dbPath                string
+	persisterFactory      DbFactoryHandler
+	numOfEpochsToKeep     uint32
+	numOfActivePersisters uint32
+	identifier            string
+}
+
+// NewPruningStorer will return a new instance of PruningStorer without sharded directories' naming scheme
+func NewPruningStorer(args *PruningStorerArgs) (*PruningStorer, error) {
+	return initPruningStorer(args, "")
+}
+
+// NewShardedPruningStorer will return a new instance of PruningStorer with sharded directories' naming scheme
+func NewShardedPruningStorer(
+	args *PruningStorerArgs,
+	shardId uint32,
+) (*PruningStorer, error) {
+	shardIdStr := fmt.Sprintf("%d", shardId)
+	return initPruningStorer(args, shardIdStr)
+}
+
+// initPruningStorer will create a PruningStorer with or without sharded directories' naming scheme
+func initPruningStorer(
+	args *PruningStorerArgs,
+	shardIdStr string,
+) (*PruningStorer, error) {
+	var cache storage.Cacher
+	var db storage.Persister
+	var bf storage.BloomFilter
+	var err error
+
+	defer func() {
+		if err != nil && db != nil {
+			_ = db.Destroy()
+		}
+	}()
+
+	if args.NumOfActivePersisters < 1 {
+		return nil, storage.ErrInvalidNumberOfPersisters
+	}
+	if check.IfNil(args.Notifier) {
+		return nil, storage.ErrNilEpochStartNotifier
+	}
+	if check.IfNil(args.PersisterFactory) {
+		return nil, storage.ErrNilPersisterFactory
+	}
+
+	cache, err = storageUnit.NewCache(args.CacheConf.Type, args.CacheConf.Size, args.CacheConf.Shards)
```

Comment:
storageService.GetCacheFromConfig ?

Reply:
no, it only returns the config which is expected for this constructor `NewCache`